tests/h/session_test.py

```diff
@@ -274,27 +265,17 @@ def __init__(self, authority, userid, user_authority, public_groups,
         self.route_url = mock.Mock(return_value='/group/a')
         self.session = mock.Mock(get_csrf_token=lambda: '__CSRF__')
 
-        self._authority_group_service = FakeAuthorityGroupService(public_groups)
-        self._profile_group_service = mock.Mock(spec_set=['sort'])
-        self._profile_group_service.sort = mock.Mock(side_effect=lambda groups: groups)
-
-    def set_groups(self, groups):
-        self.user.groups = groups
+        self._list_groups_service = mock.create_autospec(ListGroupsService, spec_set=True, instance=True)
```

Comment:
I've cleaned up the bits of tests in `session_test` that touch on the changes in this PR, but note that `session_test` is still a scary beast...too scary to tackle in one go here.